pos_accounting/src/pool/tests/delta_tests.rs

```diff
@@ -0,0 +1,336 @@
+            .expect("get_pool_balance ok")
+            .expect("get_pool_balance some"),
+        pledged_amount
+    );
+    assert_eq!(
+        PoSAccountingView::get_pool_data(&delta, pool_id)
+            .expect("get_pool_data ok")
+            .expect("get_pool_data some"),
+        PoolData::new(pub_key, pledged_amount)
+    );
+    assert!(delta.data().delegation_balances.data().is_empty());
+    assert!(delta.data().delegation_data.data().is_empty());
+    assert!(delta.data().pool_delegation_shares.data().is_empty());
+
+    match undo {
+        PoSAccountingUndo::CreatePool(u) => delta.undo_create_pool(u).unwrap(),
+        _ => unreachable!(),
+    }
+    assert!(delta.is_empty());
+}
+
+#[test]
+fn check_decommission_pool() {
+    let mut storage = InMemoryPoSAccounting::new();
+    let db = PoSAccountingDBMut::new_empty(&mut storage);
+    let mut delta = PoSAccountingDelta::new(&db);
+
+    let pledged_amount = Amount::from_atoms(100);
+    let (_, pub_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+    let outpoint = OutPoint::new(OutPointSourceId::BlockReward(Id::new(H256::random())), 0);
+    let (pool_id, _) = delta.create_pool(&outpoint, pledged_amount, pub_key.clone()).unwrap();
+
+    let undo = delta.decommission_pool(pool_id).unwrap();
+
+    assert!(delta.is_empty());
+
+    match undo {
+        PoSAccountingUndo::DecommissionPool(u) => delta.undo_decommission_pool(u).unwrap(),
+        _ => unreachable!(),
+    }
+
+    assert_eq!(
+        PoSAccountingView::get_pool_balance(&delta, pool_id)
+            .expect("get_pool_balance ok")
+            .expect("get_pool_balance some"),
+        pledged_amount
+    );
+    assert_eq!(
+        PoSAccountingView::get_pool_data(&delta, pool_id)
+            .expect("get_pool_data ok")
+            .expect("get_pool_data some"),
+        PoolData::new(pub_key, pledged_amount)
+    );
+    assert!(delta.data().delegation_balances.data().is_empty());
+    assert!(delta.data().delegation_data.data().is_empty());
+    assert!(delta.data().pool_delegation_shares.data().is_empty());
+}
+
+#[test]
+fn check_delegation_id() {
+    let mut storage = InMemoryPoSAccounting::new();
+    let db = PoSAccountingDBMut::new_empty(&mut storage);
+    let mut delta = PoSAccountingDelta::new(&db);
+
+    let pledged_amount = Amount::from_atoms(100);
+    let (pool_id, _) = create_pool(&mut delta, pledged_amount).unwrap();
+
+    let (_, del_pub_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+    let del_outpoint = OutPoint::new(OutPointSourceId::BlockReward(Id::new(H256::random())), 0);
+    let (delegation_id, undo) =
+        delta.create_delegation_id(pool_id, del_pub_key.clone(), &del_outpoint).unwrap();
+
+    assert_eq!(
+        delta
+            .get_delegation_data(delegation_id)
+            .expect("get_delegation_data ok")
+            .expect("get_delegation_data some"),
+        DelegationData::new(pool_id, del_pub_key)
+    );
+
+    match undo {
```

Comment:
Why not test undos?

Reply:
not sure what you mean.  Undo for `create_delegation_id` is called on line 253

Reply:
You're not checking the full state after reversal. Just that reversal will remove the data you added. Am I missing something? Please check the other tests as well.